src/parser/src/UnaryExprNode.cpp

```diff
@@ -1,97 +1,72 @@
 #include "UnaryExprNode.hpp"
 
 #include "CodeGenUtils.hpp"
-#include "MatrixNode.hpp"
 
+/**
+ * Code generation for unary elements.
+ * We only consider each element as a single value and apply the operation as a function
+ * This is based off the Binary operation code with the subtraction of the right hand side
+ * of the expression being required, as LLVM handles this for us.
+ * @param context
+ * @return pointer to the resultant matrix initial position of the result
+ */
 llvm::Value* AST::UnaryExprNode::codeGen(Utils::IRContext* context) {
-    // opval should be an evaluated matrix for which we can create a new matrix
-    llvm::Value* opVal = this->operand->codeGen(context);
     // We go through and apply the relevant unary operator to each element of
     // the matrix
-    auto newMatAlloc = Utils::createMatrix(context, *this->type);
-    // We generate the operations sequentially
-    // TODO: Add Kernel call for nvptx
-    //    recursiveUnaryGeneration(op, context->module, context->Builder, ty, newMatAlloc, opVal, dimension);
-    return opVal;
-}
+    llvm::Value* operand = this->operand->codeGen(context);
 
-/**
- * Generates LLVM IR for the Unary Expression via a recursive pass over each
- * element of the matrix in a column major systematic search. This will output
- * code that NVPTX should be able to identifiy and use to generate PTX compliant
- * code optimised for CUDA.
- *
- * Note, final two paramaters, index and prevDim can be excluded as they are
- * used within the function to recursively generate the index offset. By default
- * they are set to the identity element (1) and will not impact the result.
- * @param op
- * @param module
- * @param Builder
- * @param ty
- * @param matAlloc
- * @param opVal
- * @param dimension
- * @param index
- * @param prevDim
- */
-void AST::UnaryExprNode::recursiveUnaryGeneration(const UNA_OPERATORS& op, llvm::Module* module,
-                                                  llvm::IRBuilder<>* Builder, llvm::Type* ty,
-                                                  llvm::AllocaInst* matAlloc, llvm::Value* opVal,
-                                                  std::vector<int> dimension, int index, int prevDim) {
-    // Store of a type for the matrix (only need this for the final pass)
-    llvm::ArrayType* matType;
-    if (dimension.size() == 1) {
-        matType = llvm::ArrayType::get(ty, index * dimension.at(0));
-    }
+    auto operandMatNode = std::dynamic_pointer_cast<AST::ExprNode>(this->operand);
+    llvm::Value* matAlloc;
```

Comment:
This should be `llvm::Value* matAlloc {};` to prevent uninitialised return.

Reply:
Done